clash-prelude/src/Clash/Class/BitPack/BitShift.hs

```diff
@@ -0,0 +1,50 @@
+{-|
+Copyright  :  (C) 2025, QBayLogic B.V.
+License    :  BSD2 (see the file LICENSE)
+Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+-}
+
+{-# LANGUAGE Safe #-}
+
+{-# OPTIONS_GHC -fplugin GHC.TypeLits.Normalise #-}
+
+module Clash.Class.BitPack.BitShift where
+
+import Clash.Class.BitPack.Internal (BitPack (..), bitCoerce)
+import Clash.Promoted.Nat           (SNat (..))
+import Clash.Sized.BitVector        (BitVector)
+
+{- $setup
+>>> import Clash.Prelude
+-}
+
+{-# INLINE (<<%) #-}
+-- | Left shift the bits by the given amount.
+--
+-- >>> (1 :: Unsigned 3) <<% d0
+-- 1
+-- >>> (1 :: Unsigned 3) <<% d2
+-- 4
+-- >>> (1 :: Unsigned 3) <<% d4
+-- 0
+-- >>> pack (-1 :: Signed 6)
+-- 0b11_1111
+-- >>> reduceAnd (-1 :: Signed 6)
+infixl 7 <<%
+(<<%) :: forall a n. BitPack a => a -> SNat n -> a
```

Comment:
This is too general and should be limited to `Bits a`. It makes no sense to add operators that can shift an `Either Bool (Maybe Int)` value.

Reply:
I agree with this perspective @christiaanb. Also, I don't think it makes sense to go through a `BitVector` conversion for things that already have an efficient simulation path through `Bits`.